app/controlplane/internal/service/workflow.go

```diff
@@ -123,7 +124,15 @@ func (s *WorkflowService) View(ctx context.Context, req *pb.WorkflowServiceViewR
 		return nil, err
 	}
 
-	wf, err := s.useCase.FindByIDInOrg(ctx, currentOrg.ID, req.Id)
+	var wf *biz.Workflow
+	if req.Name != nil {
+		wf, err = s.useCase.FindByNameInOrg(ctx, currentOrg.ID, *req.Name)
+	} else if req.Id != nil {
+		wf, err = s.useCase.FindByIDInOrg(ctx, currentOrg.ID, *req.Id)
+	} else {
+		return nil, errors.BadRequest("invalid", "either workflow ID or Name is required")
```

Comment:
as I mentioned we could add an API validation

Reply:
Done!